tools/test-proxy/Azure.Sdk.Tools.TestProxy/RecordingHandler.cs

```diff
@@ -27,11 +37,332 @@ public RecordingHandler(string targetDirectory)
             RepoPath = targetDirectory;
         }
 
+        private static readonly RecordedTestSanitizer defaultSanitizer = new RecordedTestSanitizer();
+
+        private static readonly string[] s_excludedRequestHeaders = new string[] {
+            // Only applies to request between client and proxy
+            // TODO, we need to handle this properly, there are tests that actually test proxy functionality.
+            "Proxy-Connection",
+        };
+
+        // Headers which must be set on HttpContent instead of HttpRequestMessage
+        private static readonly string[] s_contentRequestHeaders = new string[] {
+            "Content-Length",
+            "Content-Type",
+        };
+
+        public List<RecordedTestSanitizer> Sanitizers = new List<RecordedTestSanitizer>
+        {
+            new RecordedTestSanitizer()
+        };
+
+        public List<ResponseTransform> Transforms = new List<ResponseTransform>
+        {
+            new StorageRequestIdTransform(),
+            new ClientIdTransform()
+        };
+
+        public readonly ConcurrentDictionary<string, (string File, ModifiableRecordSession ModifiableSession)> recording_sessions
+            = new ConcurrentDictionary<string, (string, ModifiableRecordSession)>();
+
+        private readonly ConcurrentDictionary<string, ModifiableRecordSession> playback_sessions 
+            = new ConcurrentDictionary<string, ModifiableRecordSession>();
+
+        public RecordMatcher Matcher = new RecordMatcher();
+        #endregion
+
+        #region recording functionality
+        public void StopRecording(string id, bool saveToDisk)
+        {
+            if (!recording_sessions.TryRemove(id, out var fileAndSession))
+            {
+                return;
+            }
+
+            if (saveToDisk)
+            {
+                var (file, session) = fileAndSession;
+
+                foreach (RecordedTestSanitizer sanitizer in session.AdditionalSanitizers.Count > 0 ? Sanitizers.Concat(session.AdditionalSanitizers) : Sanitizers)
```

Comment:
```suggestion
                foreach (RecordedTestSanitizer sanitizer in Sanitizers.Concat(session.AdditionalSanitizers))
```

Reply:
I actually avoided doing this specifically to avoid always allocating a new array into memory. (it actually was originally exactly like this :P).